pro/src/components/StocksEventList/StocksEventList.tsx

```diff
@@ -135,32 +142,41 @@ const StocksEventList = ({
       (stock) => !stocksIdToDelete.includes(stock.id)
     )
 
-    const deletedStocksCount = stocks.length - newStocks.length
     logEvent?.(Events.CLICKED_BULK_DELETE_STOCK, {
       offerId: offerId,
-      deletionCount: `${stocks.length - newStocks.length}`,
+      deletionCount: stocksIdToDelete.length,
     })
-    setIsCheckedArray(stocks.map(() => false))
     setStocks?.([...newStocks])
+    setIsCheckedArray(stocks.map(() => false))
+
     if (stocksIdToDelete.length > 0) {
-      await Promise.all(
-        stocksIdToDelete.map(
-          (stockId) => stockId?.toString() && api.deleteStock(stockId)
+      // If all stocks are checked without any stock unchecked,
+      // use the delete all route with filters
+      if (areAllChecked) {
```

Comment:
hum en fait révélation :
```suggestion
      if (areAllChecked && hasNoFilters) {
```

nan ?
vérifie avec Martin ptet le comportement
je filtre, je coche tout sélectionner -> je supprime quoi ? tout Mais avec le filtre non ?

Reply:
Cf l'appel api, on envoie les filtres courants au back et le back delete tous les stocks correspondants aux filtres

Reply:
ah oui en effet !